packages/server/src/ratings.ts

```diff
@@ -0,0 +1,75 @@
+import {
+    GameResponse,
+    AbstractGame,
+    UserRanking,
+  } from "@ogfcommunity/variants-shared";
+  import { Glicko2, Player } from "glicko2";
+  import { updateUserRating, getUser } from "./users";
+  
+  const ranking = new Glicko2({
+    tau: 0.5,
+    rating: 1500,
+    rd: 350,
+    vol: 0.06,
+  });
+  
+  const matches: [Player, Player, number][] = [];
+  
+  export async function updateRatings(
+    game: GameResponse,
+    game_obj: AbstractGame<object, object>,
+  ) {
+    // Rating system currently only for two player games
+    const winner = game_obj.result[0] === "B" ? 1 : 0;
```

Comment:
Is this correct?  Usually "B" is player `0`

<img width="210" alt="Screenshot 2024-11-15 at 5 26 12 PM" src="https://github.com/user-attachments/assets/81a018fa-0c98-439c-9117-6482a27635d2">

Also, some 2-player variants (like chess) don't follow this pattern, but it's fine to ignore this for now.

Reply:
Yes this is correct. This `matches.push([player1, player2, winner]);` function this indicates that if winner is 1 then player1 won over player2. Since I pass player_black into the function first I am saying that player_black won/lost over player_white so "B" can be 1 or 0

Reply:
Ah, I see.  In that case, I think this variable is misleading (seems to say "Player X is the winner" when the opposite is true).  Maybe we call it `outcome`? This is what the glicko2 library names it: https://github.com/mmai/glicko2js/blob/master/glicko2.d.ts#L41


Also, should we handle ties? Quantum supports them: https://github.com/govariantsteam/govariants/blob/0abf1b20636ce4d87ed3b3e378d598a527affebb/packages/shared/src/variants/quantum.ts#L117